### DISCUSSION TITLE
next/router first render my component, and then push the new path

### DISCUSSION THREAD

**Discussion by FernandoTimo** [vercel/next.js#15987] [Help]
Please how can i correct this behavior?
what i hope is that it redirects before loading the component, i really appreciate your answer.
```javascript
import { useRouter } from 'next/router';
import { useContext, useEffect } from 'react';
import Auth from 'context/Authorization';
export default function Dashboard() {
  const router = useRouter();
  const { isAutorized, setIsAuthorized } = useContext(Auth);
  useEffect(() => {
    if (isAutorized === false) {
      router.push('/');
    }
  }, []);
  return (
    <>
      {isAutorized ? (
        <>
          <MainHead />
          <Navigation />
          <Body>
            <Section>
              <Content c>
                <h1 className={styleDashboard.Title}>Dashboard</h1>
              </Content>
            </Section>
          </Body>
        </>
      ) : (
        '🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥'
      )}
    </>
  );
}
```

**Comment by haggen** [comment]
In the simplest form you could do an earlier return:

```
export default function Dashboard() {
  // ...

  if (!isAuthorized) {
    return <h1>Loading...</h1>;
  }

  return (
    // ...
  );
}
```

Another approach would be via an earlier component in the tree or even a HOC that would conditionally render your component.

When you're using a hook in the same component though it's technically impossible to prevent the component to render because otherwise the hook wouldn't have been invoked. You can only control what the component will render. React rendering process runs parallel to the side effects (such as `useEffect` hook). So you cannot condition one on the other on the same run. You have to wait for the next render to do something like that. Does that make sense?
